DataFormats/EgammaReco/interface/ElectronSeed.h

```diff
@@ -23,93 +47,127 @@
-       float dPhi1 =std::numeric_limits<float>::infinity() ) ;
-
+    void setCaloCluster( const CaloClusterRef& clus){caloCluster_=clus;isEcalDriven_=true;}
+    void addHitInfo(const PMVars& hitVars){hitInfo_.push_back(hitVars);}
+    void setNrLayersAlongTraj(int val){nrLayersAlongTraj_=val;}
     //! Accessors
     const CtfTrackRef& ctfTrack() const { return ctfTrack_ ; }
     const CaloClusterRef& caloCluster() const { return caloCluster_ ; }
-    unsigned char hitsMask() const { return hitsMask_ ; }
-    int subDet2() const { return subDet2_ ; }
-    float dRz2() const { return dRz2_ ; }
-    float dPhi2() const { return dPhi2_ ; }
-    float dRz2Pos() const { return dRz2Pos_ ; }
-    float dPhi2Pos() const { return dPhi2Pos_ ; }
-    int subDet1() const { return subDet1_ ; }
-    float dRz1() const { return dRz1_ ; }
-    float dPhi1() const { return dPhi1_ ; }
-    float dRz1Pos() const { return dRz1Pos_ ; }
-    float dPhi1Pos() const { return dPhi1Pos_ ; }
-    float hoe1() const { return hcalDepth1OverEcal_ ; }
-    float hoe2() const { return hcalDepth2OverEcal_ ; }
-
+   
     //! Utility
     TrackCharge getCharge() const { return startingState().parameters().charge() ; }
 
     bool isEcalDriven() const { return isEcalDriven_ ; }
     bool isTrackerDriven() const { return isTrackerDriven_ ; }
 
-  private:
+    const std::vector<PMVars>& hitInfo()const{return hitInfo_;}
+    float dPhiNeg(size_t hitNr)const{return getVal(hitNr,&PMVars::dPhiNeg);}
+    float dPhiPos(size_t hitNr)const{return getVal(hitNr,&PMVars::dPhiPos);}
+    float dPhiBest(size_t hitNr)const{return bestVal(dPhiNeg(hitNr),dPhiPos(hitNr));}
+    float dRZPos(size_t hitNr)const{return getVal(hitNr,&PMVars::dRZPos);}
+    float dRZNeg(size_t hitNr)const{return getVal(hitNr,&PMVars::dRZNeg);}
+    float dRZBest(size_t hitNr)const{return bestVal(dRZNeg(hitNr),dRZPos(hitNr));}
+    int detId(size_t hitNr)const{return hitNr<hitInfo_.size() ? hitInfo_[hitNr].detId : 0;}
+    int subDet(size_t hitNr)const{return DetId(detId(hitNr)).subdetId();}
+    int layerOrDiskNr(size_t hitNr)const{return getVal(hitNr,&PMVars::layerOrDiskNr);}
+    int nrLayersAlongTraj()const{return nrLayersAlongTraj_;}
+
+    //redundant, backwards compatible function names
+    //to be cleaned up asap
+    //no new code should use them
+    //they were created as time is short and there is less risk having
+    //the functions here rather than adapting all the function call to them in other
+    //CMSSW code 
+    float dPhi1()const{return dPhiNeg(0);}
+    float dPhi1Pos()const{return dPhiPos(0);}
+    float dPhi2()const{return dPhiNeg(1);}
+    float dPhi2Pos()const{return dPhiPos(1);}
+    float dRz1()const{return dRZNeg(0);}
+    float dRz1Pos()const{return dRZPos(0);}   
+    float dRz2()const{return dRZNeg(1);}
+    float dRz2Pos()const{return dRZPos(1);}   
+    int subDet1()const{return subDet(0);}
+    int subDet2()const{return subDet(1);}
+    int hitsMask()const; 
```

Comment:
Yes, so looking into it. It doesnt make any difference if its a char or unsigned char there as we are dealing with the bit representation. I was worried about the later conversion to a size_t.  However I would like to have it const unsigned char just to be consistent.  Also I found that function is a bit buggy and my testing didnt reveal it so will fix it